src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -650,50 +650,76 @@ public static string ConvertPathToDocumentUri(string path)
+                    docUriStrBld.Replace("%5C", "/");
+
+                    // Find the first colon after the "file:///" prefix, skipping the first char after
+                    // the prefix since a Windows path cannot start with a colon. End the check at
+                    // less than docUriStrBld.Length - 2 since we need to look-ahead two characters.
+                    for (int i = prefixLen + 1; i < docUriStrBld.Length - 2; i++)
                     {
                         if ((docUriStrBld[i] == '%') && (docUriStrBld[i + 1] == '3') && (docUriStrBld[i + 2] == 'A'))
                         {
                             int driveLetterIndex = i - 1;
                             char driveLetter = char.ToLowerInvariant(docUriStrBld[driveLetterIndex]);
-                            docUriStrBld.Replace(path[driveLetterIndex], driveLetter, driveLetterIndex, 1);
+                            docUriStrBld.Replace(docUriStrBld[driveLetterIndex], driveLetter, driveLetterIndex, 1);
                             break;
                         }
                     }
                 }
+                else
+                {
+                    // This is a Windows path without a drive specifier, must be either a relative or UNC path.
+                    int prefixLen = docUriStrBld.Length;
+
+                    docUriStrBld.Append(escapedPath);
+
+                    // Uri.EscapeDataString goes a bit far, encoding \ chars. Also, VSCode wants / instead of \.
+                    docUriStrBld.Replace("%5C", "/");
 
-                // Uri.EscapeDataString goes a bit far, encoding \ chars. Also, VSCode wants / instead of \.
-                docUriStrBld.Replace("%5C", "/");
+                    // The proper URI form for a UNC path is file://server/share.  In the case of a UNC
+                    // path, remove the path's leading // because the file:// prefix already provides it.
+                    if ((docUriStrBld.Length > prefixLen + 1) &&
+                        (docUriStrBld[prefixLen] == '/') &&
+                        (docUriStrBld[prefixLen + 1] == '/'))
+                    {
+                        docUriStrBld.Remove(prefixLen, 2);
+                    }
+                }
             }
             else
             {
-                // Because we will prefix later with file:///, remove the initial encoded / if this is an absolute path.
-                // See https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.7.2#implicit-file-path-support
-                // Uri.EscapeDataString goes a bit far, encoding / chars.
-                docUriStrBld.Replace("%2F", string.Empty, 0, 3).Replace("%2F", "/");
+                // On non-Windows system, simply append the escaped path.
+                docUriStrBld.Append(escapedPath);
             }
 
-            // ' is not always encoded.  I've seen this in Windows PowerShell.
+#if !CoreCLR
```

Comment:
Does this IFDEF still work since we moved to .NET Standard?

Reply:
I'm guessing not.  But at the time we do that, we'll have mulitple usages of #if CoreCLR to figure out.  I guess we could do a runtime check to see if we're using the .NET Framework.

Reply:
Yeah a runtime check probably makes the most sense and still is backport-able.

Reply:
Done